index.js

```diff
@@ -219,4 +219,33 @@ myForm.addEventListener("submit", (event) => {
         errorMessage.classList.remove("d-none");
         event.preventDefault();
     }
-});
\ No newline at end of file
+});
+
+let inputs = Array.from(document.getElementsByClassName("input"));
+
+let formInputValues = {
+    firstname:"",
+    lastname:"",
+    email: "",
+    message: "",
+};
+
+if (!localStorage.getItem('form')) {
+    localStorage.setItem('form', JSON.stringify(formInputValues));
+}else {
+    formInputValues = JSON.parse(localStorage.getItem('form'))
```

Comment:
In order to be used after retrived from the local storage the saved string is turned back into an object using `JSON.parse` or similar methods 📝 Well done!